pkg/signature/verifier.go

```diff
@@ -0,0 +1,78 @@
+//
+// Copyright 2021 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package signature
+
+import (
+	"context"
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/ed25519"
+	"crypto/rsa"
+	"io/ioutil"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+	"github.com/sigstore/sigstore/pkg/cryptoutils"
+)
+
+type Verifier interface {
+	VerifySignature(signature []byte, message []byte, opts ...VerifierOption) error
+}
+
+type VerifyRequest struct {
```

Comment:
I'm not happy that custom implementations of signers/verifiers would require type casting in order to allow for expanded options, but I also don't know of a better alternative /shrug